SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs

```diff
@@ -171,47 +191,50 @@ private static int CreateDeployment(string[] args, bool useChinaPlatform)
+                // we are using the main deployment snapshot also for the sim player deployment(s), because we only need to specify a snapshot
                 // to be able to start the deployment. The sim players don't care about the actual snapshot.
-                var createSimDeploymentOp = CreateSimPlayerDeploymentAsync(deploymentServiceClient,
-                    projectName, assemblyName, runtimeVersion, mainDeploymentName, simDeploymentName,
+                var simDeploymentCreationOps = CreateSimPlayerDeploymentsAsync(deploymentServiceClient,
+                    projectName, assemblyName, runtimeVersion, mainDeploymentName, simDeploymentBaseName,
                     simDeploymentJson, mainDeploymentSnapshotPath, simDeploymentRegion, simDeploymentCluster,
-                    simNumPlayers, useChinaPlatform);
+                    numSimPlayers, numSimDeployments, useChinaPlatform);
 
-                // Wait for both deployments to be created.
-                Console.WriteLine("Waiting for deployments to be ready...");
+                if (simDeploymentCreationOps == null || simDeploymentCreationOps.Count == 0)
+                {
+                    Console.WriteLine("Failed to start any simulated player deployments.");
+                    return 1;
+                }
+
+                // Wait for the main deployment to be ready
+                Console.WriteLine("Waiting for the main deployment to be ready...");
                 var mainDeploymentResult = createMainDeploymentOp.PollUntilCompleted().GetResultOrNull();
                 if (mainDeploymentResult == null)
                 {
                     Console.WriteLine("Failed to create the main deployment");
                     return 1;
                 }
-
                 Console.WriteLine("Successfully created the main deployment");
-                var simPlayerDeployment = createSimDeploymentOp.PollUntilCompleted().GetResultOrNull();
-                if (simPlayerDeployment == null)
+
+                // Waiting for the simulated player deployment(s) to be ready
+                var numSuccessfullyStartedSimDeployments = 0;
+                for (var simDeploymentIndex = 0; simDeploymentIndex < simDeploymentCreationOps.Count; simDeploymentIndex++)
                 {
-                    Console.WriteLine("Failed to create the simulated player deployment");
-                    return 1;
-                }
+                    Console.WriteLine($"Waiting for the simulated player deployment to be ready... " +
+                        $"(deployment {simDeploymentIndex + 1}/{simDeploymentCreationOps.Count})");
 
-                Console.WriteLine("Successfully created the simulated player deployment");
+                    var simPlayerDeployment = simDeploymentCreationOps[simDeploymentIndex].PollUntilCompleted().GetResultOrNull();
```

Comment:
Won't this block on the first one?

Reply:
All of them would begin their startup at https://github.com/spatialos/UnrealGDK/pull/2398/files#diff-893453b6043e557819612fff6168a107R196 and run it concurrently.

But yeah, it will afterwards block on them one-by-one and sequentially add flags. So the total time taken would be at most `time_taken_by_slowest_deployment + num_deployments * time_to_update_flags`.